app/changelogmodel.cpp

```diff
@@ -0,0 +1,121 @@
+#include "ChangelogModel.h"
+#include <QNetworkReply>
+#include <QXmlStreamReader>
+#include <QDebug>
+
+ChangelogModel::ChangelogModel( QObject *parent ) : QAbstractListModel{parent}
+{
+  _networkManager = new QNetworkAccessManager( this );
+  connect( _networkManager, &QNetworkAccessManager::finished, this, &ChangelogModel::onFinished );
+}
+
+void ChangelogModel::onFinished( QNetworkReply *reply )
+{
+  if ( reply->error() == QNetworkReply::NoError )
+  {
+    QXmlStreamReader xml( reply );
+    QString title, description, link, pubDate;
+    while ( !xml.atEnd() )
+    {
+      xml.readNext();
+      if ( xml.isStartElement() )
+      {
+        if ( xml.name().toString() == "item" )
+        {
+          title.clear();
+          description.clear();
+          link.clear();
+          pubDate.clear();
+        }
+        else if ( xml.name().toString() == "title" )
+        {
+          title = xml.readElementText();
+        }
+        else if ( xml.name().toString() == "description" )
+        {
+          description = xml.readElementText();
+        }
+        else if ( xml.name().toString() == "link" )
+        {
+          link = xml.readElementText();
+        }
+        else if ( xml.name().toString() == "pubDate" )
+        {
+          pubDate = xml.readElementText();
+        }
+      }
+      else if ( xml.isEndElement() )
+      {
+        if ( xml.name().toString() == "item" )
+        {
+          QDateTime dt = QDateTime::fromString( pubDate, "ddd, dd MMM yyyy hh:mm:ss t" );
+          if ( dt > _lastSeen )
+          {
+            beginInsertRows( QModelIndex(), rowCount(), rowCount() );
+            _logs << Changelog{ title, description, link, dt };
+            endInsertRows();
+          }
+        }
+      }
+    }
+    if ( xml.hasError() )
+    {
+      qWarning() << "XML error:" << xml.errorString();
+    }
+  }
+  else
+  {
+    qWarning() << "Network error:" << reply->errorString();
+  }
+  reply->deleteLater();
+
+  if ( !_logs.isEmpty() )
+  {
+    emit dataChanged( createIndex( 0, 0 ), createIndex( rowCount(), 0 ) );
+  }
+}
+
+QHash<int, QByteArray> ChangelogModel::roleNames() const
+{
+  return
+  {
+    { TitleRole, "title" },
```

Comment:
shouldn't be these translated too?

Reply:
No need, it is just registering these roles to use in QML